torch/csrc/cuda/nccl.cpp

```diff
@@ -99,6 +96,13 @@ ncclDataType_t to_nccl_data_type(const at::Tensor& t) {
   }
 }
 
+ncclDataType_t to_nccl_data_type(const at::Tensor& t) {
+  if (!t.is_cuda()) {
+    throw std::runtime_error("Unconvertible NCCL type");
```

Comment:
nit: this is prior to this PR, shall we be more explicit on the error message? Should this be the following? 

```
f"NCCL only supports CUDA tensors, but got a tensor on {t.device}"
```